### ISSUE TITLE
Python, Ruby: use protoGenPackageDeps

### ISSUE THREAD

**Issue by jmuk** [googleapis/packman#135]
`--proto_gen_pkg_dep` flag has been added by @garrettjonesgoogle for mixin support, but it's not yet used for other langauges. Especially Python and Ruby package template should use them for additional dependencies.

**Comment by jmuk** [comment]
Cc: @geigerj

**Comment by landrito** [comment]
Would a simple map between proto_gen_pkg_dep to package name for each language suffice or should more sophisticated logic be used?

**Comment by garrettjonesgoogle** [comment]
proto_gen_pkg_dep is just supposed to refer to the api_name used in the consumed package. In general, this just means prefixing "grpc-" (for Java).

**Comment by landrito** [comment]
Ah I see. This is the case for [Python](https://github.com/googleapis/api-client-staging/blob/master/generated/python/grpc-google-iam-v1/setup.py#L19) and [Ruby](https://github.com/googleapis/api-client-staging/blob/master/generated/ruby/grpc-google-iam-v1/grpc-google-iam-v1.gemspec#L5) as well. 

So should the assumption be to prefix with "grpc-" unless it is google-common-protos or otherwise specified?

**Comment by garrettjonesgoogle** [comment]
Yeah. FYI in Java, we even do the same thing for google-common-protos :-)

**Comment by landrito** [comment]
Gotcha! Thanks for the clarification!
